pkg/apis/application/v1alpha1/types.go

```diff
@@ -966,6 +968,9 @@ func (app *Application) CascadedDeletion() bool {
 
 func (app *Application) IsRefreshRequested() (RefreshType, bool) {
 	refreshType := RefreshTypeNormal
+	if app.Spec.Alwayshardrefresh {
+		refreshType = RefreshTypeHard
+	}
 	annotations := app.GetAnnotations()
 	if annotations == nil {
 		return refreshType, false
```

Comment:
I'm not sure what you mean here. I have to explicitly overwrite the pre-defined variable refreshType. If I define refreshType in a if-statement, golang will throw undefined variable build errors, since neither option in the if-statement might be used.

Reply:
Tested your suggested change on line 972 with
`return RefreshTypeHard, true`
It results in a permanent refresh-loop every 1-2 seconds